provider/pkg/provider/provider.go

```diff
@@ -1129,8 +1129,13 @@ func (p *cfnProvider) Delete(ctx context.Context, req *pulumirpc.DeleteRequest)
 			return nil, errors.Wrapf(err, "failed to parse inputs for update")
 		}
 
+		// Retrieve the state.
+		state, err := plugin.UnmarshalProperties(req.GetProperties(), plugin.MarshalOptions{
+			Label: fmt.Sprintf("%s.state", label), KeepUnknowns: true, SkipNulls: true, KeepSecrets: true,
```

Comment:
Curious, preferring SkipNulls intentionally? 

Also, KeepUnknowns does not matter as there are no unknowns in state I think.

Reply:
This is to stay consistent with how we're reading the state in other life-cycle methods (e.g. Read: https://github.com/pulumi/pulumi-aws-native/blob/2962615c932c7ea3d791f8500acb2bf285d69251/provider/pkg/provider/provider.go#L919C2-L921C4)